jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactory.java

```diff
@@ -148,65 +145,66 @@ private SslConnectionFactory newSslConnectionFactory(Map<Path, FileChannel> keyS
         }
         if (!sslConfig.excludeCipherSuite().isEmpty()) {
-            final String[] ciphs = new String[sslConfig.excludeCipherSuite().size()];
+            String[] ciphs = new String[sslConfig.excludeCipherSuite().size()];
             for (int i = 0; i < ciphs.length; i++) {
                 ciphs[i] = sslConfig.excludeCipherSuite(i).name();
             }
             factory.setExcludeCipherSuites(ciphs);
 
         }
         if (!sslConfig.includeCipherSuite().isEmpty()) {
-            final String[] ciphs = new String[sslConfig.includeCipherSuite().size()];
+            String[] ciphs = new String[sslConfig.includeCipherSuite().size()];
             for (int i = 0; i < ciphs.length; i++) {
                 ciphs[i] = sslConfig.includeCipherSuite(i).name();
             }
             factory.setIncludeCipherSuites(ciphs);
-
         }
 
-
-        Optional<String> password = Optional.of(sslConfig.keyDbKey()).
-                filter(key -> !key.isEmpty()).map(secretStore::getSecret);
-
+        Optional<String> keyDbPassword = secret(sslConfig.keyDbKey());
         switch (sslConfig.keyStoreType()) {
             case PEM:
                 factory.setKeyStore(getKeyStore(sslConfig.pemKeyStore(), keyStoreChannels));
-                if (password.isPresent()) {
+                if (keyDbPassword.isPresent())
                     log.warning("Encrypted PEM key stores are not supported.");
-                }
                 break;
             case JKS:
                 factory.setKeyStorePath(sslConfig.keyStorePath());
                 factory.setKeyStoreType(sslConfig.keyStoreType().toString());
-                factory.setKeyStorePassword(password.orElseThrow(passwordRequiredForJKSKeyStore("key")));
+                factory.setKeyStorePassword(keyDbPassword.orElseThrow(passwordRequiredForJKSKeyStore("key")));
                 break;
         }
 
         if (!sslConfig.trustStorePath().isEmpty()) {
             factory.setTrustStorePath(sslConfig.trustStorePath());
-            factory.setTrustStoreType(sslConfig.trustStoreType().toString());
-            factory.setTrustStorePassword(password.orElseThrow(passwordRequiredForJKSKeyStore("trust")));
+            factory.setTrustStoreType(sslConfig.trustStoreType().toString());      
+            if (sslConfig.useTrustStorePassword())
+                factory.setTrustStorePassword(keyDbPassword.orElseThrow(passwordRequiredForJKSKeyStore("trust")));
         }
 
         factory.setKeyManagerFactoryAlgorithm(sslConfig.sslKeyManagerFactoryAlgorithm());
         factory.setProtocol(sslConfig.protocol());
         return new SslConnectionFactory(factory, HttpVersion.HTTP_1_1.asString());
     }
 
+    /** Returns the secret password with the given name, or empty if the password name is null or empty */
+    private Optional<String> secret(String keyname) {
+        return Optional.of(keyname).filter(key -> !key.isEmpty()).map(secretStore::getSecret);
```

Comment:
I prefer having _filter_ and _map_ operations on separate lines to improve readability, but this is okay as the lambda operations are simple.